docs/DeveloperGuide.md

```diff
@@ -1,29 +1,134 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Save / Withdraw feature
+
+The save and withdraw feature is implemented based on the Wallet class data structure. 
+The Wallet class has several components: the username, the password, the default currency, the total balance, 
+and the deposit list containing different types of currency and their balances.
+
+Given below is the UML diagram of the Wallet class.
+
+![avatar](/property/WalletClass.jpg)
+
+Here is an example usage scenario and how the save/withdraw feature behaves at each step.
+
+Step 1: The user register a new account and a new `Wallet` object is initialized.
+For now, the deposit list is empty.
+
+![avatar](/property/ObjectStep1.jpg)
+
+Step 2: The user login and executes `save sgd 100` to save 100 singapore dollars in the wallet. 
+The `save` command first parses the command and verify its validity. 
+After that, it calls the `saveMoney` method of the `Wallet` class, 
+and the `saveMoney` method traverses the deposit list and finds that there's no existed SGD deposit. 
+Therefore, it creates a new `Deposit` object with the currency of SGD and the balance of 100, 
+and then adds it to the deposit list.
+Also, the total balance is updated.
+
+![avatar](/property/ObjectStep2.jpg)
+
+Step 3: The user executes `withdraw sgd 20` to withdraw 20 singapore dollars in the wallet. 
+The `withdraw` command parsed the command and verify its validity. 
+Afterwards, it calls the `withdrawMoney` method of the `Wallet` class, 
+and the `withdrawMoney` method traverses the deposit list and finds that there's existing SGD deposit. 
+Therefore, it finds the corresponding `Deposit` object 
+and calls the `withdraw` method within it to withdraw 20 singapore dollars.
+Similarly, the total balance is updated.
+
+![avatar](/property/ObjectStep3.jpg)
+
+### Add/remove personal currency feature 
+
+The personal currency feature allows users to add a new currency to the list of currencies that is stored on their computer. They also have the opportunity to remove that same currency from the list so that it can’t be used anymore, but they are not able to remove any other currencies, that have been preset when downloading the jar.
+
+Given below are the UML diagrams of the classes that are needed for adding/removing a personal currency: 
+
+![avatar](/property/newcurrency1.png)
+
+Step 1: The user calls the command new_currency and is then prompted in the CLI with the following commands: 
+Please enter your new currency's abbreviation: `btc`
+Please enter your new currency's full name: `bitcoin`
+Please enter your new currency's symbol: `bc`
+Please enter your new currency's rate: `0.000049`
+
+Based on that a new currency is created (in this example we have created a new currency of bitcoin).
+
+![avatar](/property/newcurrency2.png)
```

Comment:
Header should be :CurrencyStructure since this is an object instance, not the class.